src/Strategy/CommonClassesStrategy.php

```diff
@@ -91,4 +96,9 @@ public static function getCandidates($type)
 
         return [];
     }
+
+    public static function buzzInstantiate()
+    {
+        return new \Buzz\Client\FileGetContents(MessageFactoryDiscovery::find());
```

Comment:
can buzz work with any message factory? there is the super edge case where somebody is using discovery to find a client but then instantiates their own message implementations so we can't find a message factory.

as buzz is the last client in order, that seems irrelevant to me, however. the hyper-correct thing would be to add a condition callback function above that makes sure message factory discovery can find a message factory...

Reply:
Yes, Buzz supports PSR17 and Httplug message factories. 

Hm, that is true. 

Btw, buzz does not have "their own factories". So it would be quite normal that someone installed buzz and not having any message factories installed.

Reply:
> the hyper-correct thing would be to add a condition callback function above that makes sure message factory discovery can find a message factory...

I agree with you. Then I spend 20 minutes on implementation and I realized that we dont do this for any other client. See: 
https://github.com/php-http/curl-client/blob/master/src/Client.php#L79

Reply:
:man_facepalming: sorry about that. yeah lets not, then.